backend/src/v5/services/filesManager.js

```diff
@@ -66,20 +69,38 @@ FilesManager.removeAllFilesFromModel = async (teamspace, model) => {
 };
 
 FilesManager.getFileAsStream = async (teamspace, collection, fileName) => {
-	const { type, link, size } = await getRefEntry(teamspace, collection, fileName);
+	const refEntry = await getRefEntry(teamspace, collection, fileName);
+
+	if (!refEntry) {
+		throw templates.fileNotFound;
+	}
 
 	let readStream;
-	switch (type) {
+	switch (refEntry.type) {
 	case 'fs':
-		readStream = await FSHandler.getFileStream(link);
+		readStream = await FSHandler.getFileStream(refEntry.link);
 		break;
 	case 'gridfs':
-		readStream = await GridFSHandler.getFileStream(teamspace, collection, link);
+		readStream = await GridFSHandler.getFileStream(teamspace, collection, refEntry.link);
 		break;
 	default:
-		logger.logError(`Unrecognised external service: ${type}`);
+		logger.logError(`Unrecognised external service: ${refEntry.type}`);
 		throw templates.fileNotFound;
 	}
-	return { readStream, size };
+	return { readStream, size: refEntry.size };
+};
+
+FilesManager.storeFile = (teamspace, collection, data) => {
+	const type = getDefaultStorageType();
```

Comment:
As mentioned in another file, storeFile should be dealing with the refs. The filemanager user should only need to pass in a filename, and the contents